pkg/controllers/clusteroperator_controller.go

```diff
@@ -151,39 +153,30 @@ func (r *CloudOperatorReconciler) sync(ctx context.Context, config config.Operat
 	return nil
 }
 
-// applyResources will apply all resources as is to the cluster with
-// server-side apply patch and will enforce all the conflicts
+// applyResources will apply all resources as-is to the cluster, allowing adding of custom annotations and lables
 func (r *CloudOperatorReconciler) applyResources(ctx context.Context, resources []client.Object) (bool, error) {
 	updated := false
+	var err error
 
 	for _, resource := range resources {
-		resourceExisting := resource.DeepCopyObject().(client.Object)
-		err := r.Get(ctx, client.ObjectKeyFromObject(resourceExisting), resourceExisting)
-		if errors.IsNotFound(err) {
-			klog.Infof("Resource %s %q needs to be created, operator progressing...", resource.GetObjectKind().GroupVersionKind(), client.ObjectKeyFromObject(resource))
-			updated = true
-		} else if err != nil {
-			r.Recorder.Event(resource, corev1.EventTypeWarning, "Update failed", err.Error())
-			return false, err
+		switch t := resource.(type) {
```

Comment:
I'm thinking about moving this switch to 'resourceapply' package as a separate function. WDYT?

Reply:
+1